app/stv/stv.js

```diff
@@ -0,0 +1,352 @@
+        if (fraction === 0 || vote.priorities.length === 1) continue;
+
+        // Fractional transfer. We mutate the weight for these votes by a fraction
+        vote['weight'] = vote.weight * fraction;
+        // Push the mutated votes to the list of votes to be processed in the next iteration
+        nextRoundVotes.push(vote);
+      }
+    } else {
+      // Find the lowest score
+      const minScore = Math.min(
+        ...alternatives.map(
+          (alternative) => counts[alternative.description] || 0
+        )
+      );
+
+      // Find the candidates with the lowest score
+      const minAlternatives = alternatives.filter(
+        (alternative) =>
+          (counts[alternative.description] || 0) <= minScore + EPSILON
+      );
+
+      // There is a tie for eliminating candidates. Per Scottish STV we must look at the previous rounds
+      if (minAlternatives.length > 1) {
+        let reverseIteration = iteration;
+        log.push({
+          action: 'TIE',
+          description: `There are ${
+            minAlternatives.length
+          } candidates with a score of ${Number(
+            minScore.toFixed(4)
+          )} at iteration ${reverseIteration}`,
+        });
+
+        // if the minScore is 0 then we just dont bother, and just expell all the candidates with 0
+        if (minScore === 0) {
+          minAlternatives.forEach((alternative) => {
+            log.push({
+              action: 'ELIMINATE',
+              alternative: alternative,
+              minScore: Number(minScore.toFixed(4)),
+            });
+            doneAlternatives[alternative._id] = {};
+          });
+        } else {
+          while (reverseIteration > 0) {
+            // If we are at iteartion one with a tie
+            if (reverseIteration === 1) {
+              // There is nothing we can do
+              log.push({
+                action: 'TIE',
+                description:
+                  'The backward checking went to iteration 1 without breaking the tie',
+              });
+              return {
+                result: { status: 'UNRESOLVED', winners },
+                log,
+                thr,
+              };
+            }
+            // If we are not at iteration one we can enumerate backwards to see if we can find a diff
+            else {
+              reverseIteration--;
+              // Find the log object for the last iteration
+              const logObject = log.find(
+                (entry) => entry.iteration === reverseIteration
+              );
+
+              // Find the lowest score (with regard to the alternatives in the actual iteration)
+              const iterationMinScore = Math.min(
+                ...minAlternatives.map((a) => logObject.counts[a.description])
+              );
+
+              // Find the candidates (in regard to the actual iteration) that has the lowest score
+              const iterationMinAlternatives = alternatives.filter(
```

Comment:
I think we should filter `minAlternatives`, and not all `alternatives`, as I'm not sure there won't be cases where other alternatives can have lower scores in some earlier iteration than the alternatives that may be eliminated this round.

Reply:
Your correct, that was a slip-up, nice 👍🏻